packages/babel-helpers/src/helpers/maybeArrayLike.ts

```diff
@@ -0,0 +1,15 @@
+/* @minVersion 7.9.0 */
+
+import arrayLikeToArray from "./arrayLikeToArray.ts";
+
+export default function _maybeArrayLike<T>(
+  next: (arr: ArrayLike<T>, i: number) => T[] | undefined,
+  arr: ArrayLike<T>,
+  i: number,
+) {
+  if (arr && !Array.isArray(arr) && typeof arr.length === "number") {
+    var len = arr.length;
+    return arrayLikeToArray<T>(arr, i !== void 0 && i < len ? i : len);
+  }
+  return next(arr, i);
+}
```

Comment:
While we are at this, let's rename `next` to `orElse` to make it more clear what it does. It is called when `arr` is _not_ an ArrayLike.